src/prerender.ts

```diff
@@ -76,9 +78,16 @@ export async function prerender (nitro: Nitro) {
   for (let i = 0; i < 100 && routes.size; i++) {
     for (const route of Array.from(routes)) {
       const start = Date.now()
-      const error = await generateRoute(route).catch(err => err)
-      const end = Date.now()
-      nitro.logger.log(chalk.gray(`  ├─ ${route} (${end - start}ms) ${error ? `(${error})` : ''}`))
+      let end: number | null = null
+
+      try {
+        const { contents, filePath } = await generateRoute(route)
+        end = Date.now()
+        nitro.logger.log(chalk.gray(`  ├─ ${route} (${end - start}ms)`))
+        nitro.hooks.callHook('prerender:route', { route, contents, filePath })
+      } catch (error) {
+        nitro.logger.log(chalk.gray(`  ├─ ${route} (${end - start}ms) ${error ? `(${error})` : ''}`))
```

Comment:
I would also allow hook access to the errors rendered with error.

Reply:
Thanks for the feedback.

I've made the return response of the `prerender:routes` hook to be an object or an error. I also thought of introducing a hook to handle errors such as `prerender:route:error`. If you think it's cleaner to do this, I can mek the change.

Reply:
I was thinking, that by using a single hook, we can allow:

- Custom error (hooking to validate and modify context to extract an error being thrown)
- Clear error (hooking to allow still prerendering some pages with errors)

Reply:
Alright, I'm not sure I have understood correctly, but please let me know if my last changes seems good for you.